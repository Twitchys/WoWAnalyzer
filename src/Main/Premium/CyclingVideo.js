import React from 'react';
import PropTypes from 'prop-types';

class CyclingVideo extends React.PureComponent {
  static propTypes = {
    videos: PropTypes.arrayOf(PropTypes.string).isRequired,
    randomValue: PropTypes.number,
  };

  constructor(props) {
    super(props);
    const randomValue = props.randomValue !== undefined ? props.randomValue : Math.random();
    this.state = {
      current: Math.floor(randomValue * props.videos.length),
    };
    this.handleEnded = this.handleEnded.bind(this);
  }

  handleEnded() {
    this.setState({
      current: (this.state.current + 1) % this.props.videos.length,
    });
  }

  render() {
    const { videos, ...others } = this.props;
    const currentVideo = videos[this.state.current];

    return (
      <video
        autoPlay
        muted
        onEnded={this.handleEnded}
        key={currentVideo} // without this the element doesn't rerender properly and wouldn't start the next video
        {...others}
      >
        <source src={currentVideo} type="video/mp4" />
      </video>
    );
  }
}

export default CyclingVideo;
