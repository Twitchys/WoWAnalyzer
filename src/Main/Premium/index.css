.patreon-login, .github-login {
  font-size: 2em;
  border-radius: 0;
}
.patreon-login {
  background: #e6461a;
  color: #fff !important;
  border-bottom-left-radius: 3px;
}
.patreon-login:hover {
  background: #b43e1b;
}
.github-login {
  background: #fff;
  color: #000 !important;
  border-bottom-right-radius: 3px;
}
.github-login:hover {
  background: #cdcdcd;
}
