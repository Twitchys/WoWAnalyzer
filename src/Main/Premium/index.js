import React from 'react';

import PatreonIcon from 'Icons/PatreonTiny';
import GitHubMarkIcon from 'Icons/GitHubMarkLarge';
import PremiumIcon from 'Icons/Premium';

import PatreonButton from 'Main/PatreonButton';
import GithubButton from 'Main/GithubButton';

import CyclingVideo from './CyclingVideo';
import './index.css';

const INITIAL_BACKGROUNDS = [
  '7TqE3VIAU2odkmneHU', // human salute https://giphy.com/gifs/warcraft-video-games-7TqE3VIAU2odkmneHU
  '5kFzxK6ENfjrcoMgt1', // orc salute https://giphy.com/gifs/warcraft-video-games-5kFzxK6ENfjrcoMgt1
  'PoCoePEEB1EC6buvFU', // blood elf salute https://giphy.com/gifs/warcraft-video-games-PoCoePEEB1EC6buvFU
  '8cv2DoGyUClMuCKalT', // human bow https://giphy.com/gifs/warcraft-video-games-8cv2DoGyUClMuCKalT
  'fUZW0LIJhtbDnBtqkX', // orc bow https://giphy.com/gifs/warcraft-video-games-fUZW0LIJhtbDnBtqkX
  '2erqKiXvYpXeGH9XJ5', // gnome bow https://giphy.com/gifs/warcraft-video-games-2erqKiXvYpXeGH9XJ5
].map(code => `https://media.giphy.com/media/${code}/giphy.mp4`);
// const LOGIN_SUCCESSFUL_BACKGROUNDS = [
//   '1AdZe53o36pL2ifJrW', // approve https://giphy.com/gifs/warcraft-video-games-1AdZe53o36pL2ifJrW
//   '12gdy23jcbqdvqID9D', // human cheer https://giphy.com/gifs/warcraft-video-games-12gdy23jcbqdvqID9D
//   '2fRB16C8fbWsHTogzJ', // orc cheer https://giphy.com/gifs/warcraft-video-games-2fRB16C8fbWsHTogzJ
//   '55kiNDmHdIkd2NJ9L1', // ...gnome cheer  https://giphy.com/gifs/warcraft-video-games-55kiNDmHdIkd2NJ9L1
//   'cQ29kUD2CstrP0Cyyz', // blood elf dh cheer https://giphy.com/gifs/warcraft-video-games-cQ29kUD2CstrP0Cyyz
// ];
// const LOGIN_UNSUCCESSFUL_BACKGROUNDS = [
//   '4N1IhWlgeurYEqBpbd', // no https://giphy.com/gifs/warcraft-video-games-4N1IhWlgeurYEqBpbd
//   'fdGbKBJRylAQ3Gj1f8', // please https://giphy.com/gifs/warcraft-video-games-fdGbKBJRylAQ3Gj1f8
//   '1zgvoYwvgm7LnHpYp4', // blood elf dh no https://giphy.com/gifs/warcraft-video-games-1zgvoYwvgm7LnHpYp4
//   '8h0dtWORphvPgH4O20', // orc no https://giphy.com/gifs/warcraft-video-games-8h0dtWORphvPgH4O20
//   '4a4w6CzSj1t2Hl6gYy', // orc please https://giphy.com/gifs/warcraft-video-games-4a4w6CzSj1t2Hl6gYy
// ];

class Premium extends React.PureComponent {
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-lg-3 col-lg-offset-1 col-md-4 col-sm-5">
            <div className="panel">
              <div className="panel-body" style={{ padding: '0 15px' }}>
                <div
                  className="row image-background"
                  style={{ position: 'relative', paddingTop: 300, paddingBottom: 15 }}
                >
                  <div style={{ position: 'absolute', top: 0, bottom: 0, left: 0, right: 0, pointerEvents: 'none', overflow: 'hidden' }}>
                    <CyclingVideo
                      videos={INITIAL_BACKGROUNDS}
                      style={{ objectFit: 'cover', width: '100%', height: '100%' }}
                    />
                  </div>
                  <div className="col-md-12">
                    <h1>Premium sign in</h1>
                    <div className="description">
                      Sign in with your Patreon or GitHub account.
                    </div>
                  </div>
                </div>
                <div className="row">
                  <div className="col-lg-6" style={{ padding: 0 }}>
                    <a
                      href="/login/patreon"
                      className="btn btn-block patreon-login"
                    >
                      <PatreonIcon /> Patreon
                    </a>
                  </div>
                  <div className="col-lg-6" style={{ padding: 0 }}>
                    <a
                      href="/login/github"
                      className="btn btn-block github-login"
                    >
                      <GitHubMarkIcon /> GitHub
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="col-lg-7 col-md-8 col-sm-7">
            <div className="panel">
              <div className="panel-heading">
                <h2>WoWAnalyzer premium</h2>
              </div>
              <div className="panel-body">
                <PremiumIcon style={{ fontSize: '6em', float: 'right', color: '#e45a5a', marginTop: 0 }} />
                Help out development and get WoWAnalyzer premium in return! An early adopter gets the following rewards:<br />
                <ul>
                  <li>our thanks.</li>
                </ul><br />

                Doesn't that sound awesome?! You too can get premium by helping out development, either by helping fund development on Patreon or by contributing something via a pull request on GitHub.<br /><br />

                <PatreonButton /> <GithubButton />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Premium;
