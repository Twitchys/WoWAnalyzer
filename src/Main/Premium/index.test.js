import React from 'react';
import { shallow } from 'enzyme';

import Premium from './index';

describe('Premium', () => {
  it('matches snapshot', () => {
    const tree = shallow((
      <Premium />
    ));
    expect(tree).toMatchSnapshot();
  });
});
