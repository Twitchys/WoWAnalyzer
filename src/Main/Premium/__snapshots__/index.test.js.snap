// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Premium matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-3 col-lg-offset-1 col-sm-4"
    >
      <div
        className="panel"
      >
        <div
          className="panel-body"
          style={
            Object {
              "padding": "0 15px",
            }
          }
        >
          <div
            className="row image-background"
            style={
              Object {
                "paddingBottom": 15,
                "paddingTop": 300,
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "pointerEvents": "none",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <CyclingVideo
                style={
                  Object {
                    "height": "100%",
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
                videos={
                  Array [
                    "https://media.giphy.com/media/7TqE3VIAU2odkmneHU/giphy.mp4",
                    "https://media.giphy.com/media/5kFzxK6ENfjrcoMgt1/giphy.mp4",
                    "https://media.giphy.com/media/PoCoePEEB1EC6buvFU/giphy.mp4",
                    "https://media.giphy.com/media/8cv2DoGyUClMuCKalT/giphy.mp4",
                    "https://media.giphy.com/media/fUZW0LIJhtbDnBtqkX/giphy.mp4",
                    "https://media.giphy.com/media/2erqKiXvYpXeGH9XJ5/giphy.mp4",
                  ]
                }
              />
            </div>
            <div
              className="col-md-12"
            >
              <h1>
                Premium sign in
              </h1>
              <div
                className="description"
              >
                Sign in with your Patreon or GitHub account.
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-lg-6"
              style={
                Object {
                  "padding": 0,
                }
              }
            >
              <a
                className="btn btn-block"
                href="/login/patreon"
                style={
                  Object {
                    "background": "#e6461a",
                    "borderBottomLeftRadius": 3,
                    "borderRadius": 0,
                    "color": "#fff",
                    "fontSize": "2em",
                  }
                }
              >
                <Icon />
                 Patreon
              </a>
            </div>
            <div
              className="col-lg-6"
              style={
                Object {
                  "padding": 0,
                }
              }
            >
              <a
                className="btn btn-block"
                href="/login/github"
                style={
                  Object {
                    "background": "#fff",
                    "borderBottomRightRadius": 3,
                    "borderRadius": 0,
                    "color": "#000",
                    "fontSize": "2em",
                  }
                }
              >
                <Icon />
                 GitHub
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-7 col-sm-8"
    >
      <div
        className="panel"
      >
        <div
          className="panel-heading"
        >
          <h2>
            WoWAnalyzer premium
          </h2>
        </div>
        <div
          className="panel-body"
        >
          Help out development and get WoWAnalyzer premium in return! An early adopter gets the following rewards:
          <br />
          <ul>
            <li>
              our thanks.
            </li>
          </ul>
          <br />
          Doesn't that sound awesome?! You too can get premium by helping out development, either by helping fund development on Patreon or by contributing something via a pull request on GitHub.
          <br />
          <br />
          <PatreonButton
            text="Become a Patron"
          />
           
          <GithubButton
            text="View on GitHub"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
